src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/AdjudicationCharge.kt

```diff
@@ -0,0 +1,48 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.helper.builders
+
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.AdjudicationIncidentCharge
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.AdjudicationIncidentChargeId
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.AdjudicationIncidentOffence
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.AdjudicationIncidentParty
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.AdjudicationIncidentOffenceRepository
+
+@Component
+class AdjudicationChargeBuilderFactory(
+  private val repository: AdjudicationChargeBuilderRepository,
+) {
+  fun builder(): AdjudicationChargeBuilder {
+    return AdjudicationChargeBuilder(repository)
+  }
+}
+
+@Component
+class AdjudicationChargeBuilderRepository(
+  val adjudicationIncidentOffenceRepository: AdjudicationIncidentOffenceRepository,
+) {
+  fun lookupAdjudicationOffence(code: String): AdjudicationIncidentOffence =
+    adjudicationIncidentOffenceRepository.findByCode(code)!!
+}
+
+class AdjudicationChargeBuilder(
+  private val repository: AdjudicationChargeBuilderRepository,
+) : AdjudicationChargeDsl {
+  private lateinit var adjudicationCharge: AdjudicationIncidentCharge
+
+  fun build(
+    offenceCode: String,
+    guiltyEvidence: String?,
+    reportDetail: String?,
+    incidentParty: AdjudicationIncidentParty,
+    chargeSequence: Int,
+  ): AdjudicationIncidentCharge = AdjudicationIncidentCharge(
+    id = AdjudicationIncidentChargeId(incidentParty.id.agencyIncidentId, chargeSequence),
+    incident = incidentParty.incident,
+    partySequence = incidentParty.id.partySequence,
+    incidentParty = incidentParty,
+    offence = repository.lookupAdjudicationOffence(offenceCode),
+    guiltyEvidence = guiltyEvidence,
+    reportDetails = reportDetail,
+  )
+    .also { adjudicationCharge = it }
```

Comment:
I've been thinking about this one - I'm leaning towards always saving once built for consistency.